app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,19 @@
+module Foreman
+  class Deprecation
+    #aviliable options: controller, method, info, callstack
+    #deadline_version is the last version the method could be used
+    def self.deprecation_warning(foreman_version_deadline, options = {})
+      raise Foreman::Exception, N_("Invalid version format, please enter in x.y.z") unless foreman_version_deadline.to_s.match(/\A\d[.]\d+[.]\d\z/)
+      callstack = options[:callstack] || caller
+      log_msg = "You are using a deprecated behavior, #{foreman_version_deadline} is the last version it will be available for use."
+      log_msg += " Use #{options[:controller]}##{options[:method]} instead." if options[:controller] && options[:method]
```

Comment:
There are two options, either we take off all the parameters and just leave info(though the message won't be so structured), or I can add more parameters?